dom/test/browser/isolation.js

```diff
@@ -715,6 +715,46 @@ describe('isolation', function () {
     dispose = run();
   });
 
+  it('should handle events when parent is removed and re-added', done => {
+    let clicksCount = 0;
+
+    function Child(sources) {
+      sources.DOM.select('.foo').events('click').subscribe(() => clicksCount++);
+      return {
+        DOM: Rx.Observable.of(div('.foo', ['This is foo']))
+      };
+    }
+
+    function main(sources) {
+      const child = isolate(Child)(sources);
+      // change parent key, causing it to be recreated
+      const innerDOM$ = Rx.Observable.timer(0, 120).take(3)
+        .map(x => div({key: x}, [child.DOM]));
+      return {
+        DOM: innerDOM$
+      };
+    }
+
+    const {sinks, sources, run} = Cycle(main, {
+      DOM: makeDOMDriver(createRenderTarget(), {transposition: true}),
+    });
+
+    let dispose;
+    sources.DOM.select(':root').elements().skip(1).subscribe(function (root) {
+      setTimeout(() => {
+        const foo = root.querySelector('.foo');
+        if (!foo) return;
+        foo.click();
+      }, 0);
+    });
+    setTimeout(function(){
+      assert.strictEqual(clicksCount, 3);
```

Comment:
Yeah, this is kind of tricky -- I was puzzled by it too. It's copied from the test case immediately above.

What's happening is that the `:root` `.elements()` stream is emitting an event each time the DOM is updated. The main `DOM` stream emits 3 times so we expect 3 clicks. I'd be happy to rewrite this in a more direct fashion if you'd like.